src/db/useDb.ts

```diff
@@ -0,0 +1,15 @@
+import { RequestHandler } from "express";
+import { db } from "./index.js";
+
+declare global {
+  namespace Express {
+    interface Request {
+      db: typeof db;
+    }
+  }
+}
+
+export const useDb: RequestHandler = (req, res, next) => {
+  req.db = db;
+  next();
+};
```

Comment:
This eventually needs to become a Connection Pool beacuse opening up a connection is one of the most expensive parts of any request.

Should also lock it behind a Proxy object that doesn't acquire anything from the pool until `req.db` is actually accessed, so we aren't handing out connections to API requests that didn't ask for one. For now handing every single Request _the same_ DB object is bad but that can be a discussion point for the interview and "good enough for prototyping" otherwise.